extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```
@@ -208,6 +213,29 @@ private InputRow parseNumberDataPoint(NumberDataPoint dataPoint,
       }
     });
 
+/* Typecast the SettableByteEntity<? extends ByteEntity> source into KafkaRecordEntity
+   to fetch the record. create_time has the units of epoch time. */
+    Object entity = source.getEntity();
+    Object record;
+    long timestamp;
+    long timeUnixNano = dataPoint.getTimeUnixNano();
+    try {
+      // Get the getRecord method reflectively
+      Method getRecordMethod = entity.getClass().getMethod("getRecord");
+      record = getRecordMethod.invoke(entity);
+
+      // Assuming record is a KafkaConsumerRecord (from Apache Kafka)
+      Method getTimestampMethod = record.getClass().getMethod("timestamp");
+      timestamp = (long) getTimestampMethod.invoke(record);
+
+      double delayMinutes = (double) (timestamp - (timeUnixNano / NANOS_TO_MILLIS)) / MILLIS_PER_MINUTE;
+      delayMinutes = Math.round(delayMinutes * ROUNDING_FACTOR) / ROUNDING_FACTOR;
+
+      event.put("delayed_minutes", delayMinutes);
+    }
+    catch (Exception e) {
```

Let's not swallow all the exceptions.